plugins/omprog/omprog.c

```diff
@@ -176,44 +190,42 @@ execBinary(wrkrInstanceData_t *pWrkrData, int fdStdin, int fdStdout, int fdStder
 		maxFd -= 10;
 	}
 #	endif
-	for(i = STDERR_FILENO + 1 ; i <= maxFd ; ++i) {
-		close(i);
+	for(fd = STDERR_FILENO + 1 ; fd <= maxFd ; ++fd) {
+		close(fd);
 	}
 
 	/* reset signal handlers to default */
 	memset(&sigAct, 0, sizeof(sigAct));
 	sigemptyset(&sigAct.sa_mask);
 	sigAct.sa_handler = SIG_DFL;
-	for(i = 1 ; i < NSIG ; ++i) {
-		sigaction(i, &sigAct, NULL);
+	for(sigNum = 1 ; sigNum < NSIG ; ++sigNum) {
+		sigaction(sigNum, &sigAct, NULL);
 	}
 
 	/* we need to block SIGINT, otherwise our program is cancelled when we are
 	 * stopped in debug mode.
 	 */
 	sigAct.sa_handler = SIG_IGN;
 	sigaction(SIGINT, &sigAct, NULL);
-	sigemptyset(&set);
-	sigprocmask(SIG_SETMASK, &set, NULL);
+	sigemptyset(&sigSet);
+	sigprocmask(SIG_SETMASK, &sigSet, NULL);
 
 	alarm(0);
 
-	/* finally exec child */
-	iRet = execve((char*)pWrkrData->pData->szBinary, pWrkrData->pData->aParams, environ);
-	if(iRet == -1) {
-		/* Note: this will go to stdout of the **child**, so rsyslog will never
-		 * see it except when stdout is captured. If we use the plugin interface,
-		 * we can use this to convey a proper status back!
-		 */
-		rs_strerror_r(errno, errStr, sizeof(errStr));
-		DBGPRINTF("omprog: failed to execute program '%s': %s\n",
-			  pWrkrData->pData->szBinary, errStr);
-		openlog("rsyslogd", 0, LOG_SYSLOG);
-		syslog(LOG_ERR, "omprog: failed to execute program '%s': %s\n",
-			  pWrkrData->pData->szBinary, errStr);
-	}
+	/* finally exec program */
+	execve((char*)pData->szBinary, pData->aParams, environ);
 
-	/* we should never reach this point, but if we do, we terminate */
+failed:
+	/* an error occurred: log it and exit the child process. We use the
+	 * 'syslog' system call to log the error (we cannot use LogMsg/LogError,
+	 * since these functions add directly to the rsyslog input queue).
+	 */
+	rs_strerror_r(errno, errStr, sizeof(errStr));
+	DBGPRINTF("omprog: failed to execute program '%s': %s\n",
+			pData->szBinary, errStr);
+	openlog("rsyslogd", 0, LOG_SYSLOG);
+	syslog(LOG_ERR, "omprog: failed to execute program '%s': %s\n",
+			pData->szBinary, errStr);
 	exit(1);
```

Comment:
we must never do ```exit(1)``` inside a plugin (or even core module other than startup.

The user does not expect rsyslog to die while some subactivity is having issues. I understand it was already there, but we still need to fix that ;-)

Reply:
Note this exits the child process. It is correct.